engine/class_modules/warlock/sc_warlock_affliction.cpp

```diff
@@ -691,9 +688,10 @@ struct drain_soul_t : public affliction_spell_t
   {
     affliction_spell_t::last_tick( d );
     if ( !p()->prolonged_decimation_trigger() )
-      p()->buffs.decimating_bolt->decrement();
+      p()->buffs.decimating_bolt
+          ->decrement();  // Not sure if this should be a 'reset' instead since Decimating Bolt now no longer is
```

Comment:
`decrement` will call `expire` if there's only one stack, so you can leave it here.

Reply:
The issue isn't the decrement call so much as it is how the current implementation of Decimating Bolt always adds `p()->buffs.decimating_bolt->trigger( 3, value );` three stacks of the buff.
This does make sense since Drain Soul is the only time there is an exception to that rule and by implementing some kind of buff clearing we could handle it here instead of where the buffs are applied.

Reply:
If you trigger 3 stacks on a buff with 1 max stack, you'll only have 1 stack after, not 3. So this should be fine.

Reply:
It doesn't have a max stack of one if you are not talented into Drain Soul as affliction though. 
I couldn't even find the information on the spell itself, but there was this blue post that properly explained the new expected behaviour:
https://shadowlands.wowhead.com/news=318134/upcoming-changes-for-warlocks-on-the-shadowlands-beta-soul-rot-decimating-bolt

Reply:
Is that an issue? We're talking about code that will only be executed if you do have Drain Soul. Maybe I just misunderstood your point.

Reply:
Fair point, maybe my thought on that is overreaching the scope of this pullrequest.
I thought of it as a time saver due to making further changes to the buff application for the Drain Soul-special case unnecessary while achieving the same result.